android-components/components/feature/search/src/main/java/mozilla/components/feature/search/telemetry/SerpTelemetryFetcher.kt

```diff
@@ -0,0 +1,93 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.search.telemetry
+
+import mozilla.components.support.base.log.logger.Logger
+import mozilla.components.support.ktx.android.org.json.toList
+import mozilla.components.support.remotesettings.RemoteSettingsClient
+import mozilla.components.support.remotesettings.RemoteSettingsResult
+import org.json.JSONArray
+import org.json.JSONException
+import org.json.JSONObject
+import java.io.File
+
+internal const val REMOTE_ENDPOINT_URL = "https://firefox.settings.services.mozilla.com"
+internal const val REMOTE_ENDPOINT_BUCKET_NAME = "main"
+
+/**
+ * Parse SERP Telemetry json from remote config.
+ */
+class SerpTelemetryFetcher(
+    private val rootStorageDirectory: File,
+    private val collectionName: String,
+    private val serverUrl: String = REMOTE_ENDPOINT_URL,
+    private val bucketName: String = REMOTE_ENDPOINT_BUCKET_NAME,
+) {
+    val logger = Logger("SerpTelemetryFetcher")
+    private val remoteSettingsClient: RemoteSettingsClient
+        get() = RemoteSettingsClient(
```

Comment:
Do we need a `get`ter here? Since the configuration properties are `val` and can't change, I would assume we could just initialize this during declaration. As written, we'll be creating a new object every time we access the property.

Reply:
Nope we don't. Thanks!